src/lsdb/loaders/dataframe/margin_catalog_generator.py

```diff
@@ -73,37 +73,39 @@ def create_catalog(self) -> MarginCatalog | None:
         Returns:
             Margin catalog object, or None if the margin is empty.
         """
-        ddf, ddf_pixel_map, total_rows = self._generate_dask_df_and_map()
-        margin_pixels = list(ddf_pixel_map.keys())
-        if total_rows == 0:
+        pixels, partitions = self._get_margins()
+        if len(pixels) == 0:
             return None
+        ddf, ddf_pixel_map, total_rows = self._generate_dask_df_and_map(pixels, partitions)
+        margin_pixels = list(ddf_pixel_map.keys())
         margin_catalog_info = self._create_catalog_info(total_rows)
         margin_structure = hc.catalog.MarginCatalog(margin_catalog_info, margin_pixels)
         return MarginCatalog(ddf, ddf_pixel_map, margin_structure)
 
-    def _generate_dask_df_and_map(self) -> Tuple[dd.core.DataFrame, Dict[HealpixPixel, int], int]:
+    def _get_margins(self):
```

Comment:
```suggestion
    def _get_margins(self) -> Tuple[List[HealpixPixel], List[pd.DataFrame]]:
```

Could you add a brief docstring too?